eph-es/audit.sh

```diff
@@ -0,0 +1,132 @@
+#!/bin/bash
+
+#set -x
+
+es_url=http://localhost:8080
+follow=false
+
+while getopts "fu:" option; do
+    case ${option} in
+        f )
+            follow=true
+            ;;
+        u )
+            es_url=${OPTARG}
+            ;;
+        * )
+            echo "-f      follow\n-u url      set url"
+            exit 1
+            ;;
+    esac
+done
+
+if [ "$follow" = true ] ; then
+    live_audit
+else
+    single_audit
+fi
+
+#single line version
+#cf run-task --command "curl -X GET -s http://localhost:8080/_search -H 'Content-Type: application/json' -d '{ \"size\":10000, \"query\": { \"bool\": { \"must\": [ { \"match_phrase\": { \"eventVisibility\": { \"query\": \"customer\" } } }, { \"exists\": { \"field\": \"invokedByCustomerId\" } } ], \"filter\": [ { \"match_all\": {} } ], \"should\": [], \"must_not\": [] } } }' | jq -c '.hits.hits[]._source'" eph-es
+
+single_audit () {
+
+    echo "Running Audit Dump"
+
+    query='{
+        "size":10,
+        "query": {
+            "bool": {
+                "must": [
+                    {
+                        "match_phrase": {
+                            "eventVisibility": {
+                                "query": "customer"
+                            }
+                        }
+                    },
+                    {
+                        "exists": {
+                            "field": "invokedByCustomerId"
+                        }
+                    }
+                ],
+                "filter": [
+                    {
+                        "match_all": {}
+                    }
+                ],
+                "should": [],
+                "must_not": []
+            }
+        }
+    }'
+
+    response=$(curl -X GET -s $es_url/_search?scroll=1m -H 'Content-Type: application/json' -d "$query")
+    scroll_id=$(echo "$response" | jq -r ._scroll_id)
+
+    echo "$response" | jq -c '.hits.hits[]._source'
+
+    response=$(curl -s $es_url/_search/scroll -H "Content-Type: application/json" -d "{ \"scroll\": \"1m\", \"scroll_id\": \"$scroll_id\" }")
+    scroll_id=$(echo "$response" | jq -r ._scroll_id)
+    hits_count=$(echo "$response" | jq -r '.hits.hits | length')
+
+    while [ "$hits_count" != "0" ]; do
+
+        echo "$response" | jq -c '.hits.hits[]._source'
+
+        response=$(curl -s $es_url/_search/scroll -H "Content-Type: application/json" -d "{ \"scroll\": \"1m\", \"scroll_id\": \"$scroll_id\" }")
+        scroll_id=$(echo "$response" | jq -r ._scroll_id)
+        hits_count=$(echo "$response" | jq -r '.hits.hits | length')
+    done
```

Comment:
If the `echo` line was moved to after the logic here, could we reduce the duplication here a bit? This seems like the same logic:
```suggestion
    hits_count=""

    while [ "$hits_count" != "0" ]; do
        response=$(curl -s $es_url/_search/scroll -H "Content-Type: application/json" -d "{ \"scroll\": \"1m\", \"scroll_id\": \"$scroll_id\" }")
        scroll_id=$(echo "$response" | jq -r ._scroll_id)
        hits_count=$(echo "$response" | jq -r '.hits.hits | length')
        
        echo "$response" | jq -c '.hits.hits[]._source'
    done
```